### ISSUE TITLE
Substitute from CC to QQ

### ISSUE THREAD

**Issue by antonleykin** [Macaulay2/M2#473]
Here is a simple looking but potent bug:

```
C = matrix {{1_CC}}
lift(C,RR) -- works
lift(C,QQ) -- error
sub(C,QQ) -- SIGSEGV
```

**Comment by antonleykin** [comment]
1.9.1 update: the above still don't work, also
```
i1 : sub(1.2,QQ)
stdio:1:1:(3): error: unknown engine error
```

**Comment by DanGrayson** [comment]
`sub(1.2,QQ)` should give a better error message

**Comment by antonleykin** [comment]
This one creates a bigger problem.
```
i1 : sub(matrix{{1.0}},QQ)
-- SIGSEGV
```

**Comment by mikestillman** [comment]
Fixed in commit 7dd8aaa and ones directly before that.
